core/domain/prod_validation_jobs_one_off_test.py

```diff
@@ -1704,6 +1707,104 @@ def test_invalid_collection_version_in_model_id(self):
+            objective='objective%d' % i,
+        ) for i in python_utils.RANGE(3)]
+
+        for index, collection in enumerate(collections):
+            collection.add_node('%s' % (index * 2))
+            collection.add_node('%s' % (index * 2 + 1))
+            collection_services.save_new_collection(self.owner_id, collection)
+
+        user_id_migration.AddAllUserIdsSnapshotsVerificationJob.enqueue(
+            user_id_migration.AddAllUserIdsSnapshotsVerificationJob.create_new()
+        )
+        self.process_and_flush_pending_tasks()
+
+        self.model_instance_0 = (
+            collection_models.CollectionRightsAllUsersModel.get_by_id('0'))
+        self.model_instance_1 = (
+            collection_models.CollectionRightsAllUsersModel.get_by_id('1'))
+        self.model_instance_2 = (
+            collection_models.CollectionRightsAllUsersModel.get_by_id('2'))
+
+        self.job_class = (
+            prod_validation_jobs_one_off.
+            CollectionRightsAllUsersModelAuditOneOffJob)
+
+    def test_standard_operation(self):
+        expected_output = [
+            u'[u\'fully-validated CollectionRightsAllUsersModel\', 3]']
+        run_job_and_check_output(self, expected_output)
+
+    def test_model_with_created_on_greater_than_last_updated(self):
+        self.model_instance_0.created_on = (
+            self.model_instance_0.last_updated + datetime.timedelta(days=1))
+        self.model_instance_0.put()
+        expected_output = [(
+            u'[u\'failed validation check for time field relation check '
+            'of CollectionRightsAllUsersModel\', '
+            '[u\'Entity id %s: The created_on field has a value '
+            '%s which is greater than the value '
+            '%s of last_updated field\']]') % (
+                self.model_instance_0.id,
+                self.model_instance_0.created_on,
+                self.model_instance_0.last_updated
+            ), (
+                u'[u\'fully-validated '
+                'CollectionRightsAllUsersModel\', 2]')]
+        run_job_and_check_output(self, expected_output, sort=True)
+
+    def test_model_with_last_updated_greater_than_current_time(self):
+        self.model_instance_1.delete()
+        self.model_instance_2.delete()
+        expected_output = [(
+            u'[u\'failed validation check for current time check of '
+            'CollectionRightsAllUsersModel\', '
+            '[u\'Entity id %s: The last_updated field has a '
+            'value %s which is greater than the time when the job was run\']]'
+        ) % (self.model_instance_0.id, self.model_instance_0.last_updated)]
+
+        with self.swap(datetime, 'datetime', MockDatetime13Hours), self.swap(
+            db.DateTimeProperty, 'data_type', MockDatetime13Hours):
+            update_datastore_types_for_mock_datetime()
+            run_job_and_check_output(self, expected_output, sort=True)
+
+    def test_missing_exploration_model_failure(self):
```

Comment:
Please update the tests as well to include checks for missing user models and missing user rights models.

Reply:
Done.